File: Programmers/멀쩡한 사각형.py

def solution(w, h):
    all_square = w * h
    GCD = GCD_f(w, h)
    pattern = (w + h - GCD)

    answer = all_square - pattern
    return answer


def GCD_f(x, y):
    while x != 0:
        y = y % x
        x, y = y, x

    return y

File: Programmers JS/멀쩡한 사각형.js
function solution(w, h) {
  var answer = 1;
  let gcd = GCD(w, h);
  answer =
    w > h ? (w / gcd + (h / gcd - 1)) * gcd : (h / gcd + (w / gcd - 1)) * gcd;
  return w * h - answer;
}

function GCD(a, b) {
  if (a > b) {
    let temp = b;
    b = a;
    a = temp;
  }
  while (a !== 0) {
    let q = b % a;
    b = a;
    a = q;
  }
  return b;
}
